### PR TITLE
feat: add baichuan llm support

### LINKED ISSUE #952: AI 代理 Wasm 插件对接百川大模型
https://platform.baichuan-ai.com/

### PR DESCRIPTION
**Pull request by lizzy-0323**
### Ⅰ. Describe what this PR did
Support baichuan ai llm, api documentation (https://platform.baichuan-ai.com/docs/api)

### Ⅱ. Does this pull request fix one issue?

fix #952

### Ⅳ. Describe how to verify it
**envoy.yaml:**
```yaml
admin:
  address:
    socket_address:
      protocol: TCP
      address: 0.0.0.0
      port_value: 9901
static_resources:
  listeners:
    - name: listener_0
      address:
        socket_address:
          protocol: TCP
          address: 0.0.0.0
          port_value: 10000
      filter_chains:
        - filters:
            - name: envoy.filters.network.http_connection_manager
              typed_config:
                "@type": type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager
                scheme_header_transformation:
                  scheme_to_overwrite: https
                stat_prefix: ingress_http
                # Output envoy logs to stdout
                access_log:
                  - name: envoy.access_loggers.stdout
                    typed_config:
                      "@type": type.googleapis.com/envoy.extensions.access_loggers.stream.v3.StdoutAccessLog
                # Modify as required
                route_config:
                  name: local_route
                  virtual_hosts:
                    - name: local_service
                      domains: ["*"]
                      routes:
                        - match:
                            prefix: "/"
                          route:
                            cluster: baichuan
                            timeout: 300s
                http_filters:
                  - name: baichuan
                    typed_config:
                      "@type": type.googleapis.com/udpa.type.v1.TypedStruct
                      type_url: type.googleapis.com/envoy.extensions.filters.http.wasm.v3.Wasm
                      value:
                        config:
                          name: baichuan
                          vm_config:
                            runtime: envoy.wasm.runtime.v8
                            code:
                              local:
                                filename: /etc/envoy/main.wasm
                          configuration:
                            "@type": "type.googleapis.com/google.protobuf.StringValue"
                            value: |
                              {
                                "provider": {
                                  "type": "baichuan",
                                  "apiTokens": [
                                    "sk-5e0fbd650ba0c073b11adac9926e55b9"
                                  ],
                                  "withSearchEnhance":false
                                }
                              }

                  - name: envoy.filters.http.router
  clusters:
    - name: baichuan
      connect_timeout: 30s
      type: LOGICAL_DNS
      dns_lookup_family: V4_ONLY
      lb_policy: ROUND_ROBIN
      load_assignment:
        cluster_name: baichuan
        endpoints:
          - lb_endpoints:
              - endpoint:
                  address:
                    socket_address:
                      address: api.baichuan-ai.com
                      port_value: 443
      transport_socket:
        name: envoy.transport_sockets.tls
        typed_config:
          "@type": type.googleapis.com/envoy.extensions.transport_sockets.tls.v3.UpstreamTlsContext
          "sni": "api.baichuan-ai.com"

```
Request:
```shell
curl "http://localhost:10000/v1/chat/completions"  -H "Content-Type: application/json" -d '{
  "model": "Baichuan2-Turbo",            
  "messages": [                                                     
    {                        
      "role": "user",                
      "content": "你好，你是谁？"
    }                                                                                 
  ]                                                 
}'  
```
Response:
```json
{
   "id":"chatcmpl-Mcdc9015KFRHoSV",
   "object":"chat.completion",
   "created":1716190037,
   "model":"Baichuan2-Turbo",
   "choices":[
      {
         "index":0,
         "message":{
            "role":"assistant",
            "content":"你好！我是百川大模型，是由百川智能的工程师们创造的大语言模型，我可以和人类进行自然交流、解答问题、协助创作，帮助大众轻松、普惠的获得世界知识和专业服务。如果你有任何问题，可以随时向我提问。"
         },
         "finish_reason":"stop"
      }
   ],
   "usage":{
      "prompt_tokens":6,
      "completion_tokens":52,
      "total_tokens":58
   }
}
```
### Ⅴ. Special notes for reviews
在baichuan ai的api文档中，我发现有一个api parameter是`with_search_enhance`， 这是一个`bool`类型的参数，每次打开需要消耗额外的token，我认为该参数只需要从context json中获取即可，故没有添加到文档和代码中。

### CODE REVIEW

**Comment by johnlanni** [comment]
cc  @CH3CHO

**Review by CH3CHO — CHANGES_REQUESTED** [review]
由于先合并了另外一个模型的 provider，代码有冲突了。麻烦和上面的评论一并处理一下。谢谢！

File: plugins/wasm-go/extensions/ai-proxy/README.md [hunk 1]
```diff
@@ -64,6 +64,9 @@ Azure OpenAI 所对应的 `type` 为 `azure`。它特有的配置字段如下：
 
 通义千问所对应的 `type` 为 `qwen`。它并无特有的配置字段。
 
+#### 百川智能 (Baichuan AI)
+
+百川智能所对应的 `type` 为 `baichuan` 。它并无特有的配置字段
```
[the remark this answers is not in the document]
Comment (lizzy-0323) [reply plugins/wasm-go/extensions/ai-proxy/README.md]: 已处理。

**Review by CH3CHO — APPROVED** [review]

**Comment by Suchun-sv** [comment]
> ### Ⅰ. Describe what this PR did
> Support baichuan ai llm, api documentation (https://platform.baichuan-ai.com/docs/api)
> 
> ### Ⅱ. Does this pull request fix one issue?
> fix #952
> 
> ### Ⅲ. Why don't you add test cases (unit test/integration test)?
> ### Ⅳ. Describe how to verify it
> **envoy.yaml:**
> 
> ```yaml
> admin:
>   address:
>     socket_address:
>       protocol: TCP
>       address: 0.0.0.0
>       port_value: 9901
> static_resources:
>   listeners:
>     - name: listener_0
>       address:
>         socket_address:
>           protocol: TCP
>           address: 0.0.0.0
>           port_value: 10000
>       filter_chains:
>         - filters:
>             - name: envoy.filters.network.http_connection_manager
>               typed_config:
>                 "@type": type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager
>                 scheme_header_transformation:
>                   scheme_to_overwrite: https
>                 stat_prefix: ingress_http
>                 # Output envoy logs to stdout
>                 access_log:
>                   - name: envoy.access_loggers.stdout
>                     typed_config:
>                       "@type": type.googleapis.com/envoy.extensions.access_loggers.stream.v3.StdoutAccessLog
>                 # Modify as required
>                 route_config:
>                   name: local_route
>                   virtual_hosts:
>                     - name: local_service
>                       domains: ["*"]
>                       routes:
>                         - match:
>                             prefix: "/"
>                           route:
>                             cluster: baichuan
>                             timeout: 300s
>                 http_filters:
>                   - name: baichuan
>                     typed_config:
>                       "@type": type.googleapis.com/udpa.type.v1.TypedStruct
>                       type_url: type.googleapis.com/envoy.extensions.filters.http.wasm.v3.Wasm
>                       value:
>                         config:
>                           name: baichuan
>                           vm_config:
>                             runtime: envoy.wasm.runtime.v8
>                             code:
>                               local:
>                                 filename: /etc/envoy/main.wasm
>                           configuration:
>                             "@type": "type.googleapis.com/google.protobuf.StringValue"
>                             value: |
>                               {
>                                 "provider": {
>                                   "type": "baichuan",
>                                   "apiTokens": [
>                                     "sk-5e0fbd650ba0c073b11adac9926e55b9"
>                                   ],
>                                   "withSearchEnhance":false
>                                 }
>                               }
> 
>                   - name: envoy.filters.http.router
>   clusters:
>     - name: baichuan
>       connect_timeout: 30s
>       type: LOGICAL_DNS
>       dns_lookup_family: V4_ONLY
>       lb_policy: ROUND_ROBIN
>       load_assignment:
>         cluster_name: baichuan
>         endpoints:
>           - lb_endpoints:
>               - endpoint:
>                   address:
>                     socket_address:
>                       address: api.baichuan-ai.com
>                       port_value: 443
>       transport_socket:
>         name: envoy.transport_sockets.tls
>         typed_config:
>           "@type": type.googleapis.com/envoy.extensions.transport_sockets.tls.v3.UpstreamTlsContext
>           "sni": "api.baichuan-ai.com"
> ```
> 
> Request:
> 
> ```shell
> curl "http://localhost:10000/v1/chat/completions"  -H "Content-Type: application/json" -d '{
>   "model": "Baichuan2-Turbo",            
>   "messages": [                                                     
>     {                        
>       "role": "user",                
>       "content": "你好，你是谁？"
>     }                                                                                 
>   ]                                                 
> }'  
> ```
> 
> Response:
> 
> ```json
> {
>    "id":"chatcmpl-Mcdc9015KFRHoSV",
>    "object":"chat.completion",
>    "created":1716190037,
>    "model":"Baichuan2-Turbo",
>    "choices":[
>       {
>          "index":0,
>          "message":{
>             "role":"assistant",
>             "content":"你好！我是百川大模型，是由百川智能的工程师们创造的大语言模型，我可以和人类进行自然交流、解答问题、协助创作，帮助大众轻松、普惠的获得世界知识和专业服务。如果你有任何问题，可以随时向我提问。"
>          },
>          "finish_reason":"stop"
>       }
>    ],
>    "usage":{
>       "prompt_tokens":6,
>       "completion_tokens":52,
>       "total_tokens":58
>    }
> }
> ```
> 
> ### Ⅴ. Special notes for reviews
> 在baichuan ai的api文档中，我发现有一个api parameter是`with_search_enhance`， 这是一个`bool`类型的参数，每次打开需要消耗额外的token，我认为该参数只需要从context json中获取即可，故没有添加到文档和代码中。

疑似api token泄露

**Pull request merged**